src/PointwiseFunctions/AnalyticSolutions/Errors.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Tags {
+/*!
+ * \brief Prefix indicating the difference between two values, such as betwen a
+ * numeric quantity and the corresponding quantity computed from an analytic
+ * solution.
+ */
+template <typename Tag>
+struct Difference : db::PrefixTag, db::SimpleTag {
+  using type = db::const_item_type<Tag>;
+  using tag = Tag;
+};
+
+/*!
+ * \brief For each `Tag` in `TagsList`, where `TagsList` is a subset of
+ * `AnalyticTagsList`, compute the difference `Tag` - `Tags::Analytic<Tag>`.
+ */
+template <typename TagsList, typename AnalyticTagsList>
+struct DifferenceFromAnalyticCompute
+    : db::add_tag_prefix<::Tags::Difference, ::Tags::Variables<TagsList>>,
+      db::ComputeTag {
+  using argument_tags = tmpl::list<
+      Tags::Variables<TagsList>,
+      db::add_tag_prefix<Tags::Analytic, Tags::Variables<AnalyticTagsList>>>;
+  using return_type = ::Variables<db::wrap_tags_in<Tags::Difference, TagsList>>;
+
+  static constexpr void function(
+      const gsl::not_null<return_type*> error,
+      const ::Variables<TagsList>& numeric,
+      const ::Variables<db::wrap_tags_in<Tags::Analytic, AnalyticTagsList>>&
```

Comment:
again why isn't this `::Variables<AnalyticTagsList>` ?

Reply:
The compute tag for the analytic solution wraps each key in Tags::Analytic<>

Reply:
See above: I figured out how to make this change. I had to change the argument tag to `Tags::Analytic<Tags::Variables<AnalyticTagsList>>`, not just `Tags::Variables<AnalyticTagsList>`, and I had to change the test to pass `db::wrap_tags_in<Tags::Analytic, solution::tags<DataVector>>` for the second template parameter of `Tags::DifferenceFromAnalyticCompute` instead of simply passing `solution::tags<DataVector>` as before